qa/standalone/osd/osd-rep-recov-eio.sh

```diff
@@ -119,12 +119,24 @@ function rados_get_data() {
     COUNT=$(ceph pg $pgid query | jq '.info.stats.stat_sum.num_objects_repaired')
     test "$COUNT" = "1" || return 1
 
+    local object_osds=($(get_osds $poolname $objname))
+    local primary=${object_osds[0]}
+    local bad_peer=${object_osds[1]}
     inject_$inject rep data $poolname $objname $dir 0 || return 1
     inject_$inject rep data $poolname $objname $dir 1 || return 1
+    # Force primary to pull from the bad peer, so we can repair it too!
+    set_config osd $primary osd_debug_feed_pullee $bad_peer || return 1
+    rados_get $dir $poolname $objname || return 1
+
+    # Wait until automatic repair of bad peer is done
+    wait_for_clean || return 1
+
+    inject_$inject rep data $poolname $objname $dir 0 || return 1
+    inject_$inject rep data $poolname $objname $dir 2 || return 1
     rados_get $dir $poolname $objname || return 1
 
     COUNT=$(ceph pg $pgid query | jq '.info.stats.stat_sum.num_objects_repaired')
-    test "$COUNT" = "2" || return 1
+    test "$COUNT" = "3" || return 1
```

Comment:
@xiexingguo Now that I think about it, the correct result here should be 4.  Maybe we need a flush_pg_stat()?  Should you try to test with a deep-scrub before the hang test here to see if the pg goes inconsistent?  It should be active+clean because of the automatic repairing.

1. rados_get pulls (line 117) without any other errors

num_objects_repaired is 1

1. rados_get pulls (line 129) with an error but uses good peer
1. Push to bad peer
1. rados_get pulls (line 136) gets good peer so pull succeeds

num_objects_repaired is 4 (3 added)

- rados_get hangs (line 144) because all peers bad

num_objects_repaired is still 4 (0 added)

Reply:
@dzafman It takes me a few minutes to figure out the counters too :joy: 

The fact is repairing the same object (both broken on primary or one or more replicas) makes num_objects_repaired + 1. In other words, the counters are all logical, e.g., no matter how many actual copies of the object have.

Reply:
@dzafman It takes me a few minutes to figure out the counters too :joy: 

The fact is repairing the same object (both broken on primary and one or more replicas) makes num_objects_repaired (only) + 1. In other words, the counters are all logical, e.g., no matter how many actual copies of the object has.

Reply:
@xiexingguo I mixed up the tracking of num_shards_repaired.  I'm going to add to the test on master which will then need to be adjusted here.

Reply:
@dzafman Aah, glad to hear that :-)

Reply:
@xiexingguo I'm testing putting 488cd6a1 before the test commit and then merging it to look like 7c0303c2.  After your change there should be 1 additional shard repaired.

Reply:
@xiexingguo I tested just TEST_rados_get_with_eio and it passed with the 2 commits above.

../qa/run-standalone.sh "osd-rep-recov-eio.sh TEST_rados_get_with_eio"

Reply:
@dzafman Thanks!